framework/CMakeLists.txt

```diff
@@ -38,3 +38,18 @@ set_target_properties(${EXE}
     CXX_STANDARD ${aregCXXStandard}
     CXX_STANDARD_REQUIRED ON
     )
+
+# GTest setup
+enable_testing()
```

Comment:
@harshit496 , thank you for changes. It looks better some notes:

1. This code should not be in the `areg-sdk/framework/CMakeLists.txt`, but in the [areg-sdk/CMakeLists.txt](https://github.com/aregtech/areg-sdk/blob/master/CMakeLists.txt) file, which should include new `tests/CMakeLists.txt` file.
2. Please as well use `BUILD_TESTS` in [CMakeLists.txt](https://github.com/aregtech/areg-sdk/blob/master/CMakeLists.txt#L19) to enable compiling tests. Something like this (prototype):
```
 if (BUILD_TESTS)

   set(GOOGLE_TEST_BASE "${AREG_BASE}/thirdparty/googletest")
   set(TESTS_BASE "${AREG_BASE}/tests")
   set(UNIT_TEST_BASE "${TESTS_BASE}/units")
   include(${TESTS_BASE}/CMakeLists.txt)
...
 ````
3. Please do not forget to check whether the codes / folder of `${GOOGLE_TEST_BASE}` exists (for example, check existence `thirdparty/googletest/CMakeLists.txt`). If not, output error message with the hint to clone `googletest` module and break compilation of unit tests.
4. All test relevant actions / setups should be done in the file `tests/CMakeLists.txt`
5. All unit test relevant actions should be done in the file `tests/units/CMakeLists.txt`
6. Please keep in mind that the folder `units` may have subfolders. For example, `units/string` to make string relevant unit tests, or `units/common` to make common setups, etc. This is optional, but may happen to avoid having huge list of sources in one directory.

The rest looks good

Reply:
@aregtech @Ali-Nasrolahi need your suggestion here. 
to address the comments above I modified the `areg-sdk/CMakeLists.txt` to have the following
```
if(BUILD_TESTS)
    set(GOOGLE_TEST_BASE "${AREG_SDK_ROOT}/thirdparty/googletest")
    if(EXISTS "${GOOGLE_TEST_BASE}/CMakeLists.txt")
        # For Windows: Prevent overriding the parent project's compiler/linker settings
        include("${GOOGLE_TEST_BASE}/CMakeLists.txt")
        set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
        set(TESTS_BASE "${AREG_SDK_ROOT}/tests")
        set(UNIT_TEST_BASE "${TESTS_BASE}/units")
        include(${TESTS_BASE}/CMakeLists.txt)
    else()
        message(SEND_ERROR "Please clone the submodules using command\n
        'git clone --recurse-submodules https://github.com/aregtech/areg-sdk.git'")
    endif()
endif()
```

although the googletest CMakeLists file (`areg-sdk/thirdparty/googletest/CMakeLists.txt`)
```
if(BUILD_GMOCK)
  add_subdirectory( googlemock )
else()
  add_subdirectory( googletest )
endif()
```
 doesn't have that relative path when CMake would execute the line `include("${GOOGLE_TEST_BASE}/CMakeLists.txt")` and hence ends up with the error

```
CMake Error at thirdparty/googletest/CMakeLists.txt:36 (add_subdirectory):
  add_subdirectory given source "googlemock" which is not an existing
  directory.
```

what is the best way to handle this?

Reply:
@Ali-Nasrolahi , can you please have a look? I'll also try to have a look today late evening.